<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>1039</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>PDA</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>logo.png</normaloff>logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="mainWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1101</width>
      <height>1011</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <widget class="QLabel" name="insertStartSymbolLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>10</y>
       <width>341</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;Start symbol:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="inputStartSymbolTextEdit">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>50</y>
       <width>440</width>
       <height>50</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(4, 72, 117);
font: 10pt &quot;DejaVu Sans Mono&quot;;
background-color: rgb(212, 232, 247);</string>
     </property>
     <property name="tabChangesFocus">
      <bool>true</bool>
     </property>
     <property name="cursorWidth">
      <number>1</number>
     </property>
     <property name="placeholderText">
      <string notr="true">ex: S</string>
     </property>
    </widget>
    <widget class="QPushButton" name="showPDAPushButton">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>230</y>
       <width>161</width>
       <height>61</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);
font: 75 16pt &quot;Cambria&quot;;
background-color:rgb(7, 119, 197);
border-radius:30px;
</string>
     </property>
     <property name="text">
      <string>Convert</string>
     </property>
    </widget>
    <widget class="QWidget" name="pdaGroup" native="true">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>310</y>
       <width>781</width>
       <height>691</height>
      </rect>
     </property>
     <widget class="QWebEngineView" name="pdaWebEngineView" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>761</width>
        <height>620</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="pdaLabel">
      <property name="geometry">
       <rect>
        <x>290</x>
        <y>20</y>
        <width>271</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;PDA Graph:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="insertAlphabetLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>110</y>
       <width>411</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;Alphabet (terminals): &lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;&amp;quot;separated by commas&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="inputAlphabetTextEdit">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>150</y>
       <width>440</width>
       <height>50</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(4, 72, 117);
font: 10pt &quot;DejaVu Sans Mono&quot;;
background-color: rgb(212, 232, 247);</string>
     </property>
     <property name="tabChangesFocus">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string notr="true">ex: a,*,+,(,)</string>
     </property>
    </widget>
    <widget class="Line" name="line">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>300</y>
       <width>1021</width>
       <height>20</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="insertRulesLabel">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>10</y>
       <width>341</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;Production rules: &lt;/span&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;&amp;quot;each on a separate line&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="inputRulesTextEdit">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>50</y>
       <width>440</width>
       <height>153</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(4, 72, 117);
font: 10pt &quot;DejaVu Sans Mono&quot;;
background-color: rgb(212, 232, 247);</string>
     </property>
     <property name="tabChangesFocus">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string>ex: E -&gt; E + T | T</string>
     </property>
    </widget>
    <widget class="QLabel" name="transitionsTitleLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>320</y>
       <width>341</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;PDA Transitions:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>360</y>
       <width>251</width>
       <height>641</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(255, 255, 255);</string>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>249</width>
        <height>639</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
      <widget class="QLabel" name="transitionsTextLabel">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>251</width>
         <height>641</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: rgb(0, 0, 0);
font: 75 12pt &quot;Times New Roman&quot;;
border-color: rgb(255, 255, 255);</string>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1100</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuShow">
    <property name="title">
     <string>Show</string>
    </property>
    <addaction name="showDemoAction"/>
   </widget>
   <widget class="QMenu" name="menuDeveloped_by">
    <property name="title">
     <string>Developed by</string>
    </property>
    <addaction name="actionYoussef_Geroge_19P9880"/>
    <addaction name="actionMostafa_Nasrat_19P4619"/>
    <addaction name="actionAnthony_Amgad_19P9880"/>
    <addaction name="actionKerrolos_Wageeh_19P3468"/>
   </widget>
   <addaction name="menuShow"/>
   <addaction name="menuDeveloped_by"/>
  </widget>
  <action name="actionYoussef_Geroge_19P9880">
   <property name="text">
    <string>Youssef George    19P9824</string>
   </property>
  </action>
  <action name="actionMostafa_Nasrat_19P4619">
   <property name="text">
    <string>Mostafa Nasrat    19P4619</string>
   </property>
  </action>
  <action name="actionAnthony_Amgad_19P9880">
   <property name="text">
    <string>Anthony Amgad  19P9880</string>
   </property>
  </action>
  <action name="actionKerrolos_Wageeh_19P3468">
   <property name="text">
    <string>Kerrolos Wageeh 19P3468</string>
   </property>
  </action>
  <action name="showParsingTableAction">
   <property name="icon">
    <iconset>
     <normaloff>parsingtable.png</normaloff>parsingtable.png</iconset>
   </property>
   <property name="text">
    <string>Parsing Table</string>
   </property>
   <property name="statusTip">
    <string>Opens SLR(1) parsing table</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="showDemoAction">
   <property name="text">
    <string>Demo</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QWebEngineView</class>
   <extends>QWidget</extends>
   <header>qwebengineview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
